apps/prairielearn/src/pages/instructorGradingJob/instructorGradingJob.html.ts

```diff
@@ -22,30 +15,31 @@ export const GradingJobRowSchema = z.object({
   user_uid: UserSchema.shape.uid,
   variant_id: VariantSchema.shape.id,
   instance_question_id: VariantSchema.shape.instance_question_id,
-  assessment: AssessmentSchema.nullable(),
-  assessment_instance: AssessmentInstanceSchema.nullable(),
-  assessment_instance_group_users: z.array(GroupUserSchema).nullable(),
+  course_instance_id: VariantSchema.shape.course_instance_id,
 });
 export type GradingJobRow = z.infer<typeof GradingJobRowSchema>;
 
 export function InstructorGradingJob({
   resLocals,
   gradingJobRow,
 }: {
-  resLocals: Record<string, any>;
+  resLocals: ResLocalsForPage['course'] | ResLocalsForPage['course-instance'];
   gradingJobRow: GradingJobRow;
 }) {
   const formatGradingJobDate = (date: Date | null) =>
     date
       ? formatDate(
           date,
-          resLocals.course_instance?.display_timezone || resLocals.course.display_timezone,
+          'course_instance' in resLocals
+            ? resLocals.course_instance.display_timezone
+            : resLocals.course.display_timezone,
           { includeMs: true },
         )
       : html`&mdash;`;
+
   const variantLink =
-    gradingJobRow.instance_question_id != null
-      ? `/pl/course_instance/${resLocals.course_instance.id}/instance_question/${gradingJobRow.instance_question_id}?variant_id=${gradingJobRow.variant_id}`
+    gradingJobRow.instance_question_id != null && gradingJobRow.course_instance_id
+      ? `/pl/course_instance/${gradingJobRow.course_instance_id}/instance_question/${gradingJobRow.instance_question_id}?variant_id=${gradingJobRow.variant_id}`
```

Comment:
Minor fix: given that this route can be served from both course and course instance routes, it is strictly more correct to pull thid course instance ID from the variant itself, not the request.